### PR TITLE
Add entity building block

### PR DESCRIPTION
**Pull request by kamilbaczek**

### CODE REVIEW

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Common/Fitnet.DomainDrivenDesign.BuildingBlocks/IDomainEvent.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+{
+    /// <summary>
+    /// Gets the unique identifier of the domain event.
+    /// </summary>
+    Guid Id { get; }
+
+    /// <summary>
+    /// Gets the timestamp when the domain event occurred.
+    /// </summary>
+    DateTime OccurredOn { get; }
```
Comment (meaboutsoftware) [inline Chapter-4-applying-tactical-domain-driven-design/Fitnet.Common/Fitnet.DomainDrivenDesign.BuildingBlocks/IDomainEvent.cs:16]: Change to OccuredAt (better from grammar perspective)

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Common/Fitnet.DomainDrivenDesign.BuildingBlocks/Entity.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿namespace EvolutionaryArchitecture.Fitnet.DomainDrivenDesign.BuildingBlocks;
+
+/// <summary>
+/// Base abstract class representing an entity in the domain model.
+/// An entity is an object that is defined by its identity and continues to exist over time.
+/// </summary>
+public abstract class Entity
+{
+    private List<IDomainEvent>? _events;
```
Comment (meaboutsoftware) [inline Chapter-4-applying-tactical-domain-driven-design/Fitnet.Common/Fitnet.DomainDrivenDesign.BuildingBlocks/Entity.cs:9]: Why not IList?

**Review by meaboutsoftware — CHANGES_REQUESTED** [review]

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Common/Fitnet.DomainDrivenDesign.BuildingBlocks/IDomainEvent.cs (hunk 1 above)
Comment (kamilbaczek) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Common/Fitnet.DomainDrivenDesign.BuildingBlocks/IDomainEvent.cs]: Nice. I will change it.

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Common/Fitnet.DomainDrivenDesign.BuildingBlocks/Entity.cs (hunk 2 above)
Comment (kamilbaczek) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Common/Fitnet.DomainDrivenDesign.BuildingBlocks/Entity.cs]: Good point! Done

**Review by meaboutsoftware — APPROVED** [review]

**Pull request merged**